packages/manifest-utils/src/getPref.ts

```diff
@@ -16,17 +16,19 @@ export function getPref (
   return `${prefix}${createVersionSpec(version, opts.pinnedVersion)}`
 }
 
-export function createVersionSpec (version: string | undefined, pinnedVersion?: PinnedVersion) {
-  if (!version) return '*'
+export function createVersionSpec (version: string | undefined, pinnedVersion?: PinnedVersion, rolling?: boolean) {
   switch (pinnedVersion ?? 'major') {
   case 'none':
     return '*'
   case 'major':
-    return `^${version}`
+    if (rolling) return '^'
+    return !version ? '*' : `^${version}`
```

Comment:
why is this line not just `^${version}`?

Reply:
Because version may not be defined, in which case `*` should be used instead (assuming that rolling is not enabled). E.g. `createVersionSpec(undefined, { pinnedVersion: 'major' })`

Previously there was a guard at the beginning of the function to handle this case.

```ts
export function createVersionSpec (version, ...) {
  if (!version) return '*'

  ...
}
```

But after introducing `rolling`, we first have to check if `rolling` was requested because now we can return a pinned version even if we do not know the current version number, e.g. `^`. Otherwise we check if `version` is defined and return a result accordingly.